orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/support/TargetServerGroup.groovy

```diff
@@ -68,9 +71,15 @@ class TargetServerGroup {
     static Location locationFromServerGroup(Map<String, Object> serverGroup) {
       // All Google server group operations currently work with zones, not regions.
       if (serverGroup.type == "gce") {
-        return new Location(type: Location.Type.ZONE, value: serverGroup.zones[0])
+        return Location.zone(serverGroup.zones[0])
+      }
+      if (serverGroup.namespace) {
+        return Location.namespace(serverGroup.namespace)
+      } else if (serverGroup.region) {
+        return Location.region(serverGroup.region)
+      } else {
+        throw new IllegalArgumentException("No known location type provided.")
```

Comment:
Could probably add more detail into this exception message (at a minimum, the serverGroup should be referenced).